src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,142 @@
+    private static RegistrationService registrationService;
+    private static StorageDao storageDao;
+    private User correctUser;
+    private User wrongUser;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        correctUser = new User();
+        correctUser.setAge(23);
+        correctUser.setLogin("david23");
+        correctUser.setPassword("1d2f3h");
+    }
+
+    @Test
+    void register_userNull_notOk() {
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(null);
+        });
+    }
+
+    @Test
+    void register_Ok() {
+        registrationService.register(correctUser);
+        assertEquals(storageDao.get(correctUser.getLogin()), correctUser);
+    }
+
+    @Test
+    void register_loginDuplicate_notOk() {
+        storageDao.add(correctUser);
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(correctUser);
+        });
+    }
+
+    @Test
+    void register_age_notOk() {
+        correctUser.setAge(17);
+        try {
+            registrationService.register(correctUser);
+        } catch (RuntimeException e) {
+            return;
+        }
+        fail("Runtime exception must be thrown if age less than 18!");
+    }
+
+    @Test
+    void register_ageIsNull_notOk() {
+        correctUser.setAge(null);
+        try {
+            registrationService.register(correctUser);
+        } catch (RuntimeException e) {
+            return;
+        }
+        fail("Runtime exception must be thrown if age is Null!");
+    }
+
+    @Test
+    void register_password_notOk() {
+        correctUser.setPassword("1212");
+        try {
+            registrationService.register(correctUser);
+        } catch (RuntimeException e) {
+            return;
+        }
+        fail("Runtime exception must be thrown if age less than 18!");
+    }
+
+    @Test
+    public void register_loginNull_notOk() {
+        correctUser.setLogin(null);
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(correctUser);
+        });
+    }
+
+    @Test
+    public void register_passwordNull_notOk() {
+        correctUser.setPassword(null);
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(correctUser);
+        });
+    }
+
+    @Test
+    void register_passwordEmpty_notOk() {
+        correctUser.setPassword("        ");
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(correctUser);
+        });
+    }
+
+    @Test
+    void register_loginEmpty_notOk() {
+        correctUser.setLogin("       ");
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(correctUser);
+        });
+    }
+
+    @Test
+    void register_password_has_spaces_notOk() {
```

Comment:
```suggestion
    void register_passwordHasSpaces_notOk() {
```
We have only 3 parts of the test method
_methodUnderTest_state_expectedBehavior_ ©README